codalab/worker_manager/kubernetes_worker_manager.py

```diff
@@ -158,6 +158,27 @@ def start_worker_job(self) -> None:
                         ],
                     }
                 ],
+                # Only one worker pod should be scheduled per node.
+                'affinity': {
+                    'podAntiAffinity': {
+                        'requiredDuringSchedulingIgnoredDuringExecution': [
+                            {
+                                'podAffinityTerm': {
+                                    'labelSelector': {
+                                        "matchExpressions": [
+                                            {
+                                                "key": "type",
+                                                "operator": "In",
+                                                "values": ["cl-worker"],
+                                            }
+                                        ]
+                                    },
+                                },
+                                'topologyKey': 'topology.kubernetes.io/zone',
```

Comment:
I'm actually not sure this topology key is correct; from the example given [here](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone), it looks to me like this word only work if each of the VMs serving as nodes were spun up in different [zones](https://cloud.google.com/compute/docs/regions-zones#zones_and_clusters) on GCP. Will that be the case?

Reply:
I'm not sure, these docs (https://blog.kubecost.com/blog/kubernetes-node-affinity) say "The topologyKey topology.kubernetes.io/zone ensures that the pod is scheduled only if the node is within the same zone as one of the existing pods with the matching key-value pairs. Note that some constraints are applied to the topology due to performance and security considerations found in the Kubernetes documentation."

Not exactly sure that that means though.

Reply:
I see. I'll double-check this. I'm just concerned that if the zone really is the GCP zone (e.g. `us-west-1c`, then we'll only be able to have a single pod on the entire cluster of nodes within that GCP zone and so this won't work... but I'll double-check

Reply:
I believe we fixed this now

Reply:
Yeah, I believe we did by changing the topology key from zone to hostname.